src/primitives/block.cpp

```diff
@@ -43,19 +43,39 @@ std::string CBlock::ToString() const
 
 uint256 DynaFedParamEntry::CalculateRoot() const
 {
-    if (IsNull()) {
+    if (m_serialize_type == 0) {
         return uint256();
     }
 
+    std::vector<uint256> compact_leaves;
+    compact_leaves.push_back(SerializeHash(m_signblockscript, SER_GETHASH, 0));
+    compact_leaves.push_back(SerializeHash(m_signblock_witness_limit, SER_GETHASH, 0));
+    uint256 compact_root(ComputeFastMerkleRoot(compact_leaves));
```

Comment:
Is this in any way different from `uint256 compact_root = ComputeFastMerkleRoot(compact_leaves));`?

Reply:
It's identical.

Reply:
Assuming the compiler is smart enough to not construct then make a copy here :)

Reply:
Cool. Personally I think the `type var = value;` syntax is a bit more readable, but that's a nit.